src/SFA.DAS.RoatpGateway.Web/Controllers/RoatpGatewayApplicationActionsController.cs

```diff
@@ -111,8 +111,12 @@ public async Task<IActionResult> ConfirmWithdrawApplication(Guid applicationId,
             }
             else if (viewModel.ConfirmApplicationAction == HtmlAndCssElements.RadioButtonValueYes)
             {
-                // This would normally do something - to be implemented
-                return View("~/Views/Gateway/ConfirmWithdrawApplication.cshtml", viewModel);
+                var username = HttpContext.User.UserDisplayName();
+                var userId = HttpContext.User.UserId();
+                await _applyApiClient.WithdrawApplication(viewModel.ApplicationId, viewModel.OptionYesText, userId, username);
+
+                viewModel.ApplicationReferenceNumber = application.ApplyData.ApplyDetails.ReferenceNumber;
```

Comment:
is reference number always there?  Not sure, add some ? if it might not be.

Reply:
it'll be there for now as you can only withdraw a submitted application